plugins/gatsby-source-wagtail/gatsby-node.js

```diff
@@ -0,0 +1,38 @@
+const fetch = require('node-fetch');
+
+exports.sourceNodes = async (
+  { actions, createNodeId, createContentDigest },
+  configOptions,
+) => {
+  const { createNode } = actions;
+  // Gatsby adds a configOption that's not needed for this plugin, delete it
+  delete configOptions.plugins; // eslint-disable-line
+  const fetchBrandingData = async () => {
+    try {
+      const response = await fetch(configOptions.pagesApiUrl);
+      return await response.json();
+    } catch (e) {
+      return console.error(`${e.name}: ${e.message}`); // eslint-disable-line
+    }
+  };
+
+  // the gatsby node shape will be the exact same as what the api returns.
+  const processNode = (node, nodeType) => createNode({
+    ...node,
+    id: createNodeId(`wagtail-node-${node.id}`),
+    parent: null,
+    children: [],
+    internal: {
+      type: nodeType,
+      contentDigest: createContentDigest(node),
+    },
+  });
+
+  const data = await fetchBrandingData();
+  if (data && data.length) {
+    return data.map(page => processNode(page, 'page'));
+  }
+  // no branding data was found
+  console.error(`No branding data was returned from ${process.env.HOST_NAME}`); // eslint-disable-line
+  return {};
```

Comment:
Is it okay that you're returning an object here but on line 33, you're returning an array? In the documentation for `sourceNodes`, I don't see them returning anything. Just curious if returning is necessary.

Reply:
Looking back at the api for it, you're right, no return needed. Not a clue why I returned an empty object there, but the first one was to prevent the error message from printing when it succeeds. Reformatted.